### DISCUSSION TITLE
Cannot send LLDP packet for d, Too big message

### DISCUSSION THREAD

**Discussion by kri164** [lldpd/lldpd#501] [Q&A] [answered]
Hello all,

what this error message means?

Tcpdump-ed ethernet packet is 200 bytes about. So I guess some TLV is too long. But how to know what?
Looking at source code https://github.com/lldpd/lldpd/blob/master/src/daemon/protocols/lldp.c dosnt make me clever.

Thank you
```
cat /etc/lldpd.d/lldpd.conf
configure system ip management pattern 10.0.0.10
configure system interface pattern eno1
configure system description Debian

lldpcli show interface
-------------------------------------------------------------------------------
LLDP interfaces:
-------------------------------------------------------------------------------
Interface:    eno1
  Administrative status: RX and TX
  Chassis:     
    ChassisID:    mac 00:11:22:33:44:55
    SysName:      debian.some.where.com
    SysDescr:     Debian
    MgmtIP:       10.0.0.10
    Capability:   Bridge, on
    Capability:   Router, off
    Capability:   Wlan, off
    Capability:   Station, off
  Port:        
    PortID:       mac 22:00:33:66:44:55
    PortDescr:    eno1
  TTL:          120
-------------------------------------------------------------------------------
```

**Comment by vincentbernat** [comment]
Unfortunately, there is a small bug that makes the interface triggering the error unreadable. But as you only have eno1, it should be this one. This should also mean that you cannot get the packet with tcpdump. Are you sure you capture the packet from lldpd?

I've never got a report about this message, but one of the cause would be that eno1 reports a MTU too low. What's the output of `ip l l dev eno1`?

**Comment by kri164** [comment]
Hello Vincent,
thank you for your help.

**ip l l dev eno1**
3: eno1: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast master vmbr0 state UP mode DEFAULT group default qlen 1000
    link/ether 64:00:6a:66:d7:a1 brd ff:ff:ff:ff:ff:ff
    altname enp0s25

Aruba switch on the other end of the wire shows my LLDP info, so I believe the data goes out of Debian Bullseye.

Im using this tcpdump options to catch packet:
**tcpdump -s0 -vv -i eno1 ether proto 0x88cc**
tcpdump: listening on eno1, link-type EN10MB (Ethernet), snapshot length 262144 bytes
14:14:17.784236 LLDP, length 92
        Chassis ID TLV (1), length 7
          Subtype MAC address (4): 04:5e:a4:1a:36:9e (oui Unknown)
          0x0000:  0404 5ea4 1a36 9e
        Port ID TLV (2), length 7
          Subtype MAC address (3): 64:00:6a:66:d7:a1 (oui Unknown)
          0x0000:  0364 006a 66d7 a1
        Time to Live TLV (3), length 2: TTL 120s
          0x0000:  0078
        System Name TLV (5), length 10: deb-01.lan
          0x0000:  6465 622d 3031 2e6c 616e
        System Description TLV (6), length 6
          Debian
          0x0000:  4465 6269 616e
        System Capabilities TLV (7), length 4
          System  Capabilities [Bridge, WLAN AP, Router, Station Only] (0x009c)
          Enabled Capabilities [Bridge] (0x0004)
          0x0000:  009c 0004
        Management Address TLV (8), length 12
          Management Address length 5, AFI IPv4 (1): 10.0.0.10
          Unknown Interface Numbering (1): 0
          0x0000:  0501 0a00 000a 0100 0000 0000
        Port Description TLV (4), length 4: eno1
          0x0000:  656e 6f31
        Organization specific TLV (127), length 9: OUI IEEE 802.3 Private (0x00120f)
          Link aggregation Subtype (3)
            aggregation status [supported], aggregation port ID 0
          0x0000:  0012 0f03 0100 0000 00
        Organization specific TLV (127), length 9: OUI IEEE 802.3 Private (0x00120f)
          MAC/PHY configuration/status Subtype (1)
            autonegotiation [supported, enabled] (0x03)
            PMD autoneg capability [10BASE-T hdx, 10BASE-T fdx, 100BASE-TX hdx, 100BASE-TX fdx, 1000BASE-T fdx] (0xec01)
            MAU type 1000BASET fdx (0x001e)
          0x0000:  0012 0f01 03ec 0100 1e
        End TLV (0), length 0

I want to achieve to stop this error message flood my system log with thousand of lines per day.

**Comment by vincentbernat** [comment]
If you are able to recompile lldpd, it would help if you could replace the line with the "Too big message" in `src/daemon/protocols/lldp.c` with the following line:

```
log_info("lldp", "Cannot send LLDP packet for %s, Too big message", hardware->h_ifname);
```

It would confirm the issue is with `eno1` and not something else.

**Comment by kri164** [comment]
OK, I will try to recompile in next week and will report back. Thank you Message ID: ***@***.***>

**Comment by kri164** [comment, marked as answer]
So, the recompiled version points to eno1.

I've noticed the eno1 is alias for enp0s25. Replacing eno1 with enp0s25 in lldpd config make result no packet goes to wire and aruba switch dont see anything.

I did try to use vmbr0 interface too, and it seems working. No errors in logs and packets go go wire. So Im happy with this.
The eno1 alias must have been created by default Proxmox installator without my knowledge.

The source code line
log_info("lldp", "Cannot send LLDP packet for %s, Too big message", hardware->h_ifname);
is very usefull and should be include in next lldp version.

Vincent, thank you very much for your quick and succesfull help.

**Comment by vincentbernat** [comment]
From the output of `ip l l`, `eno1` is the real interface name. `enp0s25` is some kind of alias, but most software, including lldpd, don't know how to handle that.

So, it is quite odd that you get the packets sent to `eno1` without issue and at the same time get a message about packet to be sent to `eno1` being too big.

With `ps auxww | grep lldpd`, you should have two lldpd process. Using the PID of the second one, could you give the output of `sudo lsof -np PID | grep RAW`. You should get only one entry.

Also, can you modify again the line with "too big message" with:

```c
	log_info("lldp", "Cannot send LLDP packet for %s, too big message %d %d", hardware->h_ifname, length, hardware->h_mtu);
```

This should give another hint of what's wrong.

**Comment by kri164** [comment]
Feb 07 14:09:03 px-03 systemd[1]: Started LLDP daemon.
Feb 07 14:09:03 px-03 lldpd[3454053]: protocol LLDP enabled
Feb 07 14:09:03 px-03 lldpd[3454053]: protocol CDPv1 disabled
Feb 07 14:09:03 px-03 lldpd[3454053]: protocol CDPv2 disabled
Feb 07 14:09:03 px-03 lldpd[3454053]: protocol SONMP disabled
Feb 07 14:09:03 px-03 lldpd[3454053]: protocol EDP disabled
Feb 07 14:09:03 px-03 lldpd[3454053]: protocol FDP disabled
Feb 07 14:09:03 px-03 lldpd[3454053]: libevent 2.1.12-stable initialized with epoll method
Feb 07 14:09:05 px-03 lldpcli[3454052]: management pattern set to new value 10.0.0.10
Feb 07 14:09:05 px-03 lldpcli[3454052]: iface-pattern set to new value eno1
Feb 07 14:09:05 px-03 lldpd[3454053]: Cannot send LLDP packet for eno1, too big message 2 1500
Feb 07 14:09:05 px-03 lldpcli[3454052]: lldpd should resume operations
Feb 07 14:09:27 px-03 pvedaemon[1485888]: <root@pam> successful auth for user 'root@pam'
Feb 07 14:09:35 px-03 lldpd[3454053]: Cannot send LLDP packet for eno1, too big message 2 1500
Feb 07 14:10:05 px-03 lldpd[3454053]: Cannot send LLDP packet for eno1, too big message 2 1500
Feb 07 14:10:35 px-03 lldpd[3454053]: Cannot send LLDP packet for eno1, too big message 2 1500
Feb 07 14:11:05 px-03 lldpd[3454053]: Cannot send LLDP packet for eno1, too big message 2 1500

My previous statement was not right. Packet is not send to wire. Switch showed cached info from previous config.

Im not continue to investigate this more, it is beyond my scope, but thank you Vincent for your help and time.
Regards

**Reply by vincentbernat** [reply]
To help debug, you can replace every `goto toobig` by `{ log_warnx("11"); goto toobig; }` (with a different string each time). It should tell when it fails.

**Comment by IAMJDA** [comment]
I've got the same problem and it's also a proxmox host 😞

The debug log:

```
2022-05-12T20:37:15 [ DBG/rpc] client asked to resume lldpd
2022-05-12T20:37:15 [ DBG/alloc] general configuration cleanup
2022-05-12T20:37:15 [ DBG/event] trigger sending PDU for port eno3
2022-05-12T20:37:15 [ DBG/send] send PDU on eno3
2022-05-12T20:37:15 [ DBG/send] fallback to protocol LLDP for eno3
2022-05-12T20:37:15 [ DBG/lldp] send LLDP PDU to eno3
2022-05-12T20:37:15 [INFO/lldp] **Cannot send LLDP packet for ▒CK, Too big message**
2022-05-12T20:37:15 [ DBG/event] trigger sending PDU for port eno4
2022-05-12T20:37:15 [ DBG/event] trigger sending PDU for port enp94s0
2022-05-12T20:37:15 [ DBG/send] send PDU on enp94s0
2022-05-12T20:37:15 [ DBG/send] fallback to protocol LLDP for enp94s0
2022-05-12T20:37:15 [ DBG/lldp] send LLDP PDU to enp94s0

```

My assumption is another one (as in my case eno3 is not an alias for enp94s0 -> "igb 0000:01:00.0 eno3: renamed from eth0"): way up in the debug log the lldpd is listing nearly 4k VLANs for eno3.

My network config part:
```
auto vmbr0
iface vmbr0 inet static
        bridge-ports eno3
        bridge-stp off
        bridge-fd 0
        bridge-vlan-aware yes
        bridge-vids 2-4094
```

Limit it to two VLANs -> no error message anymore 😁

**Reply by vincentbernat** [reply]
Good finding. We should be able to limit the number of advertised VLANs.

**Reply by vincentbernat** [reply]
This is somewhat fixed by 8e9dbbf80918.
